quil-rs/src/expression.rs

```diff
@@ -936,46 +1440,46 @@ mod tests {
         #[test]
         fn exponentiation_works_as_expected(left in arb_expr(), right in arb_expr()) {
             let expected = Expression::Infix (InfixExpression { left: Box::new(left.clone()), operator: InfixOperator::Caret, right: Box::new(right.clone()) } );
-            prop_assert_eq!(left.clone() ^ right.clone(), expected.clone());
+            prop_assert_eq!(&(left.clone() ^ right.clone()), &expected);
```

Comment:
Should be able to do this, right? I imagine the macro would automatically take the reference to each argument.

```suggestion
            prop_assert_eq!(left.clone() ^ right.clone(), expected);
```

Reply:
I thought so too, but:
1. 1.70 `clippy` yelled about me about needless `.clone` calls, but
2. removing them made tests not compile

Reply:
I'll split into two tests apiece, avoiding this issue.